hooks/tk-multi-workfiles2/scene_operation_tk-maya.py

```diff
@@ -0,0 +1,236 @@
+                               "height": indiapipeline_prefs["show_resolution"]["height"],
+                               "output": "LAYERPLACEHOLDER"})   # output is alphanumeric; replace with token <Layer>
+            except KeyError as ke:
+                # logging?
+                print "Unable to find {} in indiapipeline preferences. Not setting render defaults.".format(ke)
+            else:
+                fields.pop("extension")                         # remove ma as extension to apply default img ext
+                render_path = render_temp.apply_fields(fields).replace("LAYERPLACEHOLDER", "<Layer>")
+                self.set_render_defaults(fields, render_path, frame_sq_key)
+                self.set_vray_settings(fields, render_path, frame_sq_key)
+
+        elif operation == "save":
+            # save the current scene:
+            cmds.file(save=True)
+        elif operation == "save_as":
+            # first rename the scene as file_path:
+            cmds.file(rename=file_path)
+
+            # Maya can choose the wrong file type so
+            # we should set it here explicitly based
+            # on the extension
+            maya_file_type = None
+            if file_path.lower().endswith(".ma"):
+                maya_file_type = "mayaAscii"
+            elif file_path.lower().endswith(".mb"):
+                maya_file_type = "mayaBinary"
+
+            # save the scene:
+            if maya_file_type:
+                cmds.file(save=True, force=True, type=maya_file_type)
+            else:
+                cmds.file(save=True, force=True)
+
+        elif operation == "reset":
+            """
+            Reset the scene to an empty state
+            """
+            while cmds.file(query=True, modified=True):
+                # changes have been made to the scene
+                res = QtGui.QMessageBox.question(None,
+                                                 "Save your scene?",
+                                                 "Your scene has unsaved changes. Save before proceeding?",
+                                                 QtGui.QMessageBox.Yes|QtGui.QMessageBox.No|QtGui.QMessageBox.Cancel)
+
+                if res == QtGui.QMessageBox.Cancel:
+                    return False
+                elif res == QtGui.QMessageBox.No:
+                    break
+                else:
+                    scene_name = cmds.file(query=True, sn=True)
+                    if not scene_name:
+                        cmds.SaveSceneAs()
+                    else:
+                        cmds.file(save=True)
+
+            # do new file:
+            cmds.file(newFile=True, force=True)
+            return True
+
+    def get_render_template(self, context):
+        fields = context.as_template_fields()
+        templates = context.sgtk.templates
+        entity_type = context.entity["type"]
+
+        if entity_type == "Asset":
```

Comment:
Please replace this with:
```
template_exp = "{engine_name}_{env_name}_work_render"
template_name = self.parent.resolve_setting_expression(template_exp)
render_temp = templates[template_name]
```

Reply:
Didn't realize you could do that. Thanks!